internal/catalogmetadata/cache/cache.go

```diff
@@ -64,12 +68,18 @@ func (fsc *filesystemCache) Put(catalogName, resolvedRef string, source io.Reade
 	if errToCache == nil {
 		cacheFS, errToCache = fsc.writeFS(catalogName, source)
 	}
-	fsc.cacheDataByCatalogName[catalogName] = cacheData{
+	fsc.cacheDataByCatalogName.Store(catalogName, cacheData{
 		Ref:   resolvedRef,
 		Error: errToCache,
+	})
+
+	if errToCache != nil {
+		fsc.logger.Error(errToCache, "Cache update failed", "catalog", catalogName, "ref", resolvedRef)
```

Comment:
Would this error ultimately propagate back up to to the reconciler and be reported back to the user? If so, seems duplicative to log it since it will show up in our logs anyway. Maybe add context to the returned error instead?